### PR TITLE
ArrayAccess:: method need a return type

### PR DESCRIPTION
**Pull request by nenge123**
ArrayAccess::offsetSet($offset, $value):void
ArrayAccess::offsetExists($offset):bool
ArrayAccess::offsetUnset($offset):void
ArrayAccess::offsetGet($offset):mixed

### CODE REVIEW

**Review by cmb69 — COMMENTED** [review]
Thank you for the PR!

Given that the manual is supposed to document PHP 7 and PHP 8, adding return types would be [confusing for PHP 7 users](https://3v4l.org/CnaNe). What about adding `#[ReturnTypeWillChange]` attributes? That would [work for PHP 7 and 8](https://3v4l.org/Ae1nK).

File: language/predefined/arrayaccess.xml [hunk 1]
```diff
@@ -46,34 +46,34 @@
     <programlisting role="php">
 <![CDATA[
 <?php
+/**
+ * php8 need the return type ":void,:bool,:mixed"
+ */
```
Comment (sy-records) [inline language/predefined/arrayaccess.xml:51]: There is no need to state it here.
```suggestion

```

File: language/predefined/arrayaccess.xml [hunk 2]
```diff
@@ -46,34 +46,34 @@
...
     public function __construct() {
-        $this->container = array(
-            "one"   => 1,
-            "two"   => 2,
-            "three" => 3,
-        );
     }
-
-    public function offsetSet($offset, $value) {
+    public function offsetSet($offset, $value):void {
```
Comment (sy-records) [inline language/predefined/arrayaccess.xml:60]: Using `#[ReturnTypeWillChange]` is the best

**Review by sy-records — CHANGES_REQUESTED** [review]

**Comment by afilina** [comment]
@nenge123: There is an unaddressed comment.
@sy-records: Asked for a second review to make sure that your change request was addressed the way you expected.

File: language/predefined/arrayaccess.xml [hunk 3]
```diff
@@ -47,33 +47,30 @@
 <![CDATA[
 <?php
 class Obj implements ArrayAccess {
-    private $container = array();
-
+    public $container = array(
+        "one"   => 1,
+        "two"   => 2,
+        "three" => 3,
+    );
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:54]:
```suggestion
    public $container = [
        "one"   => 1,
        "two"   => 2,
        "three" => 3,
    ];
```

File: language/predefined/arrayaccess.xml [hunk 4]
```diff
@@ -47,33 +47,30 @@
...
         return isset($this->container[$offset]);
     }
 
-    public function offsetUnset($offset) {
+    public function offsetUnset($offset):void {
         unset($this->container[$offset]);
     }
 
-    public function offsetGet($offset) {
+    public function offsetGet($offset):mixed {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:73]: Nit:
```suggestion
    public function offsetGet($offset): mixed {
```

File: language/predefined/arrayaccess.xml [hunk 5]
```diff
@@ -47,33 +47,30 @@
...
 <?php
 class Obj implements ArrayAccess {
-    private $container = array();
-
+    public $container = array(
+        "one"   => 1,
+        "two"   => 2,
+        "three" => 3,
+    );
     public function __construct() {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:55]:
```suggestion
```

File: language/predefined/arrayaccess.xml [hunk 6]
```diff
@@ -47,33 +47,30 @@
...
-        $this->container = array(
-            "one"   => 1,
-            "two"   => 2,
-            "three" => 3,
-        );
     }
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:56]:
```suggestion
```

File: language/predefined/arrayaccess.xml [hunk 7]
```diff
@@ -47,33 +47,30 @@
...
-
-    public function offsetSet($offset, $value) {
+    public function offsetSet($offset, $value):void {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:57]:
```suggestion
    public function offsetSet($offset, $value): void {
```

File: language/predefined/arrayaccess.xml [hunk 8]
```diff
@@ -47,33 +47,30 @@
...
         if (is_null($offset)) {
             $this->container[] = $value;
         } else {
             $this->container[$offset] = $value;
         }
     }
 
-    public function offsetExists($offset) {
+    public function offsetExists($offset):bool {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:65]:
```suggestion
    public function offsetExists($offset): bool {
```

File: language/predefined/arrayaccess.xml [hunk 9]
```diff
@@ -47,33 +47,30 @@
...
         return isset($this->container[$offset]);
     }
 
-    public function offsetUnset($offset) {
+    public function offsetUnset($offset):void {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:69]:
```suggestion
    public function offsetUnset($offset): void {
```

File: language/predefined/arrayaccess.xml [hunk 10]
```diff
@@ -90,6 +87,7 @@ $obj[] = 'Append 1';
 $obj[] = 'Append 2';
 $obj[] = 'Append 3';
 print_r($obj);
+print_r($obj->container);
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:90]:
```suggestion
```

File: language/predefined/arrayaccess.xml [hunk 11]
```diff
@@ -47,33 +47,28 @@
...
-        );
-    }
-
-    public function offsetSet($offset, $value) {
+    public $container = [
+        "one"   => 1,
+        "two"   => 2,
+        "three" => 3,
+    ];
+    public function offsetSet($offset, $value): void {
```
Comment (Girgias) [inline language/predefined/arrayaccess.xml:55]:
```suggestion
    ];

    public function offsetSet($offset, $value): void {
```

**Pull request merged**